src/Persistence/MsApp/MsappArchive.cs

```diff
@@ -86,6 +86,15 @@ private class TopParentJson
 
     #region Constructors
 
+    public MsappArchive(string path, bool create, IYamlSerializationFactory yamlSerializationFactory, ILogger<MsappArchive>? logger = null)
+        : this(new FileStream(path,
+            create ? FileMode.Create : FileMode.Open,
+            create ? FileAccess.Write : FileAccess.Read,
+            FileShare.Read),
+            create ? ZipArchiveMode.Create : ZipArchiveMode.Read, leaveOpen: false, yamlSerializationFactory, logger)
+    {
+    }
```

Comment:
Can you remove this constructor and use MsappArchiveFactory instead?

Reply:
Removed the constructor in question.